sochain/README.md

```diff
@@ -0,0 +1,164 @@
+# Chainlink External Adapters to query address balance from SoChain
+
+## Configuration
+
+The adapter takes the following environment variables:
+
+- `API_TIMEOUT`: Optional timeout param, defaults to `30000`
+
+## Input Params
+
+- `endpoint`: Optional endpoint param, defaults to `balance`
+
+### Balance endpoint
+
+- `dataPath`: Optional path where to find the addresses array, defaults to `addresses`
+- `confirmations`: Optional confirmations param, defaults to `6`
+
+- `addresses`: Addresses to query
+
+  {
+
+  - `address`: Address to query
+  - `coin`: Optional currency to query, defaults to `btc`, one of `(btc|dash|doge|ltc|zec)`
+  - `chain`: Optional chain to query, defaults to `mainnet`, one of `(mainnet|testnet)`
+
+  }
+
+```json
+{
+  "id": "1",
+  "data": {
+    "addresses": [
+      {
+        "address": "3D8DJLwUXFfZvE8yJRu729MZ8uLy25SuLz",
+        "coin": "btc"
+      },
+      {
+        "address": "3EyjZ6CtEZEKyc719NZMyWaJpJG5jsVJL1",
+        "coin": "btc"
+      },
+      {
+        "address": "38bzm6nhQMFJe71jJw1U7CbgNrVNpkonZF",
+        "coin": "btc"
+      },
+      {
+        "address": "3ANaBZ6odMrzdg9xifgRNxAUFUxnReesws",
+        "coin": "btc"
+      },
+      {
+        "address": "3FFgKaYkEf1M73QtzuY9DGqC7VeM2sAQhT",
+        "coin": "btc"
+      },
+      {
+        "address": "3KTeq879YjzhqkAXzZmdapJAVC6qz5qEth",
+        "coin": "btc"
+      },
+      {
+        "address": "35ULMyVnFoYaPaMxwHTRmaGdABpAThM4QR",
+        "coin": "btc"
+      }
+    ],
+    "confirmations": 3
+  }
+}
+```
+
+### Output
+
+```json
+{
+  "jobRunID": "1",
+  "data": {
+    "result": [
+      {
+        "address": "3D8DJLwUXFfZvE8yJRu729MZ8uLy25SuLz",
+        "coin": "btc",
+        "chain": "mainnet",
+        "balance": 44900000000
+      },
+      {
+        "address": "3EyjZ6CtEZEKyc719NZMyWaJpJG5jsVJL1",
+        "coin": "btc",
+        "chain": "mainnet",
+        "balance": 9899463044
+      },
+      {
+        "address": "38bzm6nhQMFJe71jJw1U7CbgNrVNpkonZF",
+        "coin": "btc",
+        "chain": "mainnet",
+        "balance": 307499838499
+      },
+      {
+        "address": "3ANaBZ6odMrzdg9xifgRNxAUFUxnReesws",
+        "coin": "btc",
+        "chain": "mainnet",
+        "balance": 904070305884
```

Comment:
Some of our BTC indexer adapters return `"balance"` as strings, while other as numbers. We need to adapt those data providers to the same API, and my vote goes that we report all balances as strings (because big numbers).

There is also an issue where some adapters return decimal value of BTC, and other int value of SATS.

Let's update this so that every adapter returns **balance as string representing total amount of SATS** on the account.